### PR TITLE
feat: stable diffusion 2.0 with PyTorch

### PR DESCRIPTION
**Pull request by telpirion**
Adds a notebook demonstrating how to train and deploy a [Stable Diffusion 2.0](https://en.wikipedia.org/wiki/Stable_Diffusion) model using PyTorch and Vertex AI.

### CODE REVIEW

**Comment by andrewferlitsch** [comment]
<p>If we are going with the new template, the code should be commented out. See the template for the exact sequene</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752907'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752867'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752889'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752907'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752928'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewferlitsch** [comment]
<p>ditto</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752996'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752944'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Nice catch! Thanks.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752956'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Good call. Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752974'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>I have. Given the urgency for this work item, we decided to update this notebook to use the Training Service later.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419752996'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419753017'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by telpirion** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/#comment-1419753037'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb [hunk 1]
```diff
@@ -0,0 +1,1326 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    save_to_gdrive = False
</pre><p>what is this code for?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (andrewferlitsch) [inline notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <p>Consider this comment from Discord:</p><p> </p><p> </p><p>Even though the original paper recommends using 4 to 6 images, the community in Discord has found that using&nbsp;<strong>10 to 12 images</strong>&nbsp;leads to better results.Oct 26, 2022</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (andrewferlitsch) [inline notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <p>You might want to highlight you are doing mixed precision training,</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (andrewferlitsch) [inline notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <p>fp16=False ?  Hum, but you trained as mixed precision?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (andrewferlitsch) [inline notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    CUSTOM_PREDICTOR_IMAGE_URI = f"us-docker.pkg.dev/vertex-ai/prediction/pytorch-gpu.1-12:latest"

Why you name this as CUSTOM, when it is a prebuilt image?
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='76'/>

Comment (andrewferlitsch) [inline notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <p>890M parameters is not realistic to run on commodity machine and GPU.</p><p>Consider at least a n1-standard-8 and a P100</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1491/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='81'/>

Comment (telpirion) [reply notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: For Colab users.

Comment (telpirion) [reply notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: That's a good point and matches my experience. That said, we don't have a ready-to-use dataset that we can provide to customers. Do you have any suggestions?

Comment (telpirion) [reply notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: Changed.

Comment (telpirion) [reply notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: Good catch!

Comment (telpirion) [reply notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <p>Done.</p>

Comment (telpirion) [reply notebooks/official/stable_diffusion/dreambooth_stablediffusion.ipynb:1]: <p>Done.</p>

**Comment by telpirion** [comment]
Going to close & reopen

**Pull request closed without merging**